eip-0006.md

```diff
@@ -0,0 +1,324 @@
+
+Note, bootstrap actions have no **action conditions** because there is no smart contract to enforce them yet and no **data-inputs** because during bootstrapping data can be arbitrarily added by the user (no need to reference existing boxes). Furthermore, it is possible for a multi-stage protocol to have one or more bootstrap actions which allow different actors to enter the protocol potentially at different stages.
+
+Every stage, action, and the protocol itself has room for a preamble/explanation of what is going on in the given section. This allows for the spec writer to add extra comments and/or make clarifications so that it is painfully clear how the protocol is supposed to work when sharing with others.
+
+Using markdown we get the benefit of being able to hyperlink the stages in the ToC, the actions for each stage, and where each one leads to. This provides us with a reasonable interface for going through informal specifications and gaining an understanding of how the different stages are inter-linked. With both an eagle's eye view using the ToC as well as with a targeted/sequential view via following path hyperlinks, we have the ability to dig into how the parts and the whole of any given informal specification work.
+
+
+### Alternative Formatting
+
+If the list format is too crowded for inputs/outputs/conditions, the following format is available as well for more complicated protocols.
+
+```md
+### Data-Inputs 
+#### Data-Input #1
+...
+##### Data-Input #2
+...
+
+### Inputs 
+#### Input #1
+...
+#### Input #2
+...
+
+### Outputs
+#### Ouput #1
+...
+#### Ouput #2
+...
+
+### Action Conditions
+#### Condition 1
+...
+#### Condition 2
+...
+
+```
+
+This allows one to use lists within each given section, thereby allowing for much more room to go into detail about specifics. This can be useful for contracts which use several registers across several boxes in their spending conditions for example.
+
+With all of that said, let's now look at an example informal specification which uses the above scheme/format to define an NFT auction protocol.
+
+
+# Example NFT Auction Protocol
+This protocol allows a user to initiate an on-chain auction for a NFT/singleton token which accepts bids in Ergs. This informal specification is based off of the discussion/original design from [this Ergo Forum thread](https://www.ergoforum.org/t/auctions-on-ergo/122).
+
+## Stage ToC
+1. [Auction Stage](<#Stage-Auction>)
+
+
+## Action ToC
+1. [Bootstrap Auction](<#Action-Bootstrap-Auction>)
+2. [Bid On Auction](<#Action-Bid-On-Auction>)
+3. [Conclude Auction](<#Action-Conclude-Auction>)
```

Comment:
An internal hierarchy cannot be used because actions often use several inputs at different stages at the same time. Thus there is no clear way to have a hierarchy, as will be obvious in the Oracle Pool specs we release publicly soon.

As such, the actions & the stages need to be separated into their own ToC.